ibis/expr/operations/reductions.py

```diff
@@ -14,25 +13,27 @@ class Reduction(ValueOp):
     _reduction = True
 
 
+class Filterable(ValueOp):
+    where = rlz.optional(rlz.boolean)
+
+
 @public
-class Count(Reduction):
-    arg = Arg(rlz.one_of((rlz.column(rlz.any), rlz.table)))
-    where = Arg(rlz.boolean, default=None)
+class Count(Filterable, Reduction):
+    arg = rlz.one_of((rlz.column(rlz.any), rlz.table))
 
     def output_type(self):
         return functools.partial(ir.IntegerScalar, dtype=dt.int64)
 
 
 @public
-class Arbitrary(Reduction):
-    arg = Arg(rlz.column(rlz.any))
-    how = Arg(rlz.isin({'first', 'last', 'heavy'}), default=None)
-    where = Arg(rlz.boolean, default=None)
+class Arbitrary(Filterable, Reduction):
```

Comment:
@cpcloud Note that introducing the `Filterable` parent changes the previous argument order:

```python
Arbitrary(arg, how, where) -> Arbitrary(arg, where, how)
```

I can restore the previous behavior if desired, but this enables better code reuse.

Reply:
See https://github.com/ibis-project/ibis/pull/3123/commits/c5d23caf516161edc371d6d386abe7cfaf152669